libraries/ESP8266mDNS/ESP8266mDNS.cpp

```diff
@@ -261,6 +261,31 @@ bool MDNSResponder::addServiceTxt(char *name, char *proto, char *key, char *valu
   return false;
 }
 
+void MDNSResponder::clearServiceTxt(const char *name, const char *proto){
+	MDNSService* servicePtr;
+	//Find the service
+  for (servicePtr = _services; servicePtr; servicePtr = servicePtr->_next) {
+    //Checking Service names
+    if(strcmp(servicePtr->_name, name) == 0 && strcmp(servicePtr->_proto, proto) == 0){
+      //found a service name match
+      if (servicePtr->_txts != 0) {
+      	MDNSTxt * txtPtr = servicePtr->_txts;
+      	MDNSTxt * nextPtr = txtPtr->_next;
+      	delete txtPtr;
+      	txtPtr = 0;
+      	while (nextPtr !=0) {
```

Comment:
why is the service clear done with a for loop and the txt clear done with a while loop? Suggest changing this to a for loop similar as service case